### PR TITLE
[nnc] aotCompiler outputSpec support quantized outputs

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68151
* #68018
* __->__ #67711
* #67623

Differential Revision: [D32115833](https://our.internmc.facebook.com/intern/diff/D32115833)

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

File: torch/csrc/jit/mobile/nnc/aot_compiler.cpp [hunk 1]
```diff
@@ -82,6 +83,17 @@ std::unique_ptr<Function> compileMethod(
     output.sizes_ = getConstSizes(ba.buf());
     // TODO: assert the output is a buffer and not a scalar
     output.dtype_ = ba.buf()->dtype().scalar_type();
+    if (isQIntType(output.dtype_)) {
+      // Supporting only static qscale/qzero
+      output.qscale_ =
+          to<DoubleImm>(torch::jit::tensorexpr::IRSimplifier::simplify(
+                            ba.buf()->qscale()))
```
Comment (ZolotukhinM) [inline torch/csrc/jit/mobile/nnc/aot_compiler.cpp:90]: Should we simplify it whenever we create the buf, would it be cleaner (if it's possible at all)?

**Review by ZolotukhinM — APPROVED** [review]
LGTM!

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32115833).

File: torch/csrc/jit/mobile/nnc/context.cpp [hunk 2]
```diff
@@ -48,17 +48,43 @@ OutputSpec::OutputSpec(const c10::IValue& value) {
...
+    qzero_ = dict.at("qzero").toInt();
+  }
 }
 
 c10::IValue OutputSpec::serialize() const {
   c10::Dict<c10::IValue, c10::IValue> dict(
       at::StringType::get(), at::AnyType::get());
   dict.insert("sizes", sizes_);
   dict.insert("dtype", dtype_);
+  if (qscale_) {
```
Comment (ljk53) [inline torch/csrc/jit/mobile/nnc/context.cpp:64]: So looks like this won't break the backward compatibility for already compiled models.

cc: @priyaramani - as we discussed, I think we need to setup CIs to test already compiled models that we plan to run in production.

**Review by ljk53 — APPROVED** [review]

**Pull request closed without merging**